src/arizona_template_renderer.erl

```diff
@@ -99,21 +99,31 @@ render_expr(Expr, Assigns) ->
     Fun = maps:get(function, Expr),
     arizona_html:to_safe(Fun(Assigns)).
 
-render_block(Block, Assigns) ->
+render_block(Block, Assigns0) ->
+    Assigns = case maps:get(is_stateful, Block) of
+        {true, {Mod, _Fun}} ->
+            Socket0 = arizona_socket:new(Mod, Assigns0),
+            Socket = arizona_live_view:mount(Mod, Socket0),
+            arizona_socket:get_assigns(Socket);
+        false ->
+            Assigns0
+    end,
+    NormAssigns = maps:get(norm_assigns, Block),
+    ChangeableAssigns = changeable_assigns(Assigns, NormAssigns),
     ChangeableIndexes = maps:get(changeable_indexes, Block),
     Changeable = maps:get(changeable, Block),
-    Dynamic = render_changeable_indexes(ChangeableIndexes, Changeable, Assigns),
-    zip(maps:get(static, Block), Dynamic).
+    ChangeableArr = render_changeable_indexes(ChangeableIndexes, Changeable, ChangeableAssigns),
+    zip(maps:get(static, Block), ChangeableArr).
 
 changeable_assigns(Assigns, NormAssigns) ->
     maps:merge(Assigns, #{K => Fun(Assigns) || K := #{function := Fun} <- NormAssigns}).
 
-zip([S | Static], [D | Dynamic]) ->
-    [S, D | zip(Static, Dynamic)];
-zip([S | Static], []) ->
-    [S | zip(Static, [])];
-zip([], [D | Dynamic]) ->
-    [D | zip([], Dynamic)];
+zip([Static | StaticArr], [Changeable | ChangeableArr]) ->
+    [Static, Changeable | zip(StaticArr, ChangeableArr)];
+zip([Static | StaticArr], []) ->
+    [Static | zip(StaticArr, [])];
+zip([], [Changeable | ChangeableArr]) ->
+    [Changeable | zip([], ChangeableArr)];
 zip([], []) ->
     [].
```

Comment:
I've changed these variable names in favor of the [JS zip function](https://github.com/arizona-framework/arizona/pull/189/files#diff-40c800ba2e247beda1d6371754609687aae5e7f3ee959704920d9ad4af20256cR178-R186). They do the same thing, but one is on the server side, and the other is on the client.